tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -265,3 +265,28 @@ TEST(SGSparseMatrix, from_dense)
 		for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex)
 			EXPECT_EQ(sparseMatrix(featIndex,vecIndex), denseMatrix(featIndex,vecIndex));
 }
+
+TEST(SGSparseMatrix, transposed_square_matrix)
+{
+	const float64_t sparse_level=0.1;
+	const index_t number_of_features=100;
+	const index_t number_of_vectors=100;
+	const index_t rand_seed=0;
+
+	SGSparseMatrix<float64_t> sparse_matrix(number_of_features, number_of_vectors);
+	GenerateMatrix<SGSparseMatrix<float64_t> >(sparse_level, number_of_features, number_of_vectors, rand_seed, &sparse_matrix);
+
+	SGSparseMatrix<float64_t> sparse_matrix_t=sparse_matrix.get_transposed();
+
+	
+	//check dimension
+	EXPECT_EQ(sparse_matrix.num_features, sparse_matrix_t.num_vectors);
+	EXPECT_EQ(sparse_matrix.num_vectors, sparse_matrix_t.num_features);
+
+	//check contents
+	for (index_t feat_index=0; feat_index<number_of_features; ++feat_index)
+	{
+		for (index_t vec_index=0; vec_index<number_of_vectors; ++vec_index)
+			EXPECT_EQ(sparse_matrix(feat_index,vec_index), sparse_matrix_t(vec_index,feat_index));
```

Comment:
@karlnapf: Caveat here, I am following the lead of other tests in the same file. All of them use ```EXPECT_EQ``` macro. I remember us having a looong conversation about keeping PRs modular (one change at a time). Also in the previous PR, all of the aforementioned changes weren't suggested. Its better I change the macro in another PR. I will squash and commit.